include/range/v3/view/any_view.hpp

```diff
@@ -416,23 +466,39 @@ namespace ranges
 
         /// \brief A type-erased view
         /// \ingroup group-views
-        template<typename Ref, category Cat = category::input>
-        struct any_view
-          : view_facade<any_view<Ref, Cat>, unknown>
+        template<typename Ref, category Cat = category::input, typename enable = void>
+        struct any_view;
+        // at least forward
+        template<typename Ref, category Cat>
+        struct any_view<Ref, Cat, std::enable_if_t<(Cat & category::forward) == category::forward>>
+          : view_facade<any_view<Ref, Cat, void>, (Cat & category::sized) == category::sized ? finite : unknown>
         {
             friend range_access;
-            CONCEPT_ASSERT(Cat >= category::forward);
+            CONCEPT_ASSERT((Cat & category::forward) == category::forward);
 
             any_view() = default;
             template<typename Rng,
                 CONCEPT_REQUIRES_(meta::and_<
                     meta::not_<Same<detail::decay_t<Rng>, any_view>>,
                     InputRange<Rng>,
-                    meta::defer<detail::AnyCompatibleRange, Rng, Ref>>::value)>
+                    meta::defer<detail::AnyCompatibleRange, Rng, Ref>>::value &&
+                    (Cat & category::sized) == category::none)>
```

Comment:
No, Casey, you are misreading this.

I will instead suggest moving `(Cat & category::sized) == category::none` before the `meta::and_` expression so it can short circuit earlier.